src/net_processing.cpp

```diff
@@ -1722,7 +1722,7 @@ static CTransactionRef FindTxForGetData(const CTxMemPool& mempool, const CNode&
     return {};
 }
 
-void static ProcessGetData(CNode& pfrom, const CChainParams& chainparams, CConnman& connman, CTxMemPool& mempool, const std::atomic<bool>& interruptMsgProc) LOCKS_EXCLUDED(cs_main)
+void static ProcessGetData(CNode& pfrom, const CChainParams& chainparams, CConnman& connman, CTxMemPool& mempool, const std::atomic<bool>& interruptMsgProc) EXCLUSIVE_LOCKS_REQUIRED(pfrom.cs_vRecv) LOCKS_EXCLUDED(cs_main)
```

Comment:
```suggestion
static void ProcessGetData(CNode& pfrom, const CChainParams& chainparams, CConnman& connman, CTxMemPool& mempool, const std::atomic<bool>& interruptMsgProc)
    EXCLUSIVE_LOCKS_REQUIRED(!cs_main, pfrom.cs_vRecv)
```

Reply:
Thanks @hebasto! TIL. It's certainly shorter, but I'm worried people might miss the `!` if reading quickly, and think this function requires `cs_main`.

Could you share why you suggest switching from `void static` to `static void`? Is it for general cleanup?

Reply:
> Thanks @hebasto! TIL. It's certainly shorter, but I'm worried people might miss the `!` if reading quickly, and think this function requires `cs_main`.

It is not about "shorter" :)
See:
- https://clang.llvm.org/docs/ThreadSafetyAnalysis.html#excludes
- https://clang.llvm.org/docs/ThreadSafetyAnalysis.html#negative-capabilities

> Could you share why you suggest switching from `void static` to `static void`? Is it for general cleanup?

E.g., fa0572d0f3b083b4c8e2e883a66e2b198c6779f1 by @MarcoFalke